apps/guardian/src/pages/Actions/SetupActions.tsx

```diff
@@ -0,0 +1,530 @@
+import { clsx } from 'clsx';
+import {
+    type ComponentType,
+    FunctionComponent,
+    PropsWithChildren,
+    useCallback,
+    useEffect,
+    useMemo,
+    useRef,
+    useState,
+} from 'react';
+import { Modal, ProgressBar, Spinner } from 'react-bootstrap';
+import { useAtomValue, useSetAtom } from 'jotai';
+import { CcdAmount } from '@concordium/web-sdk';
+
+import Button from '~/shared/Button';
+import SuccessIcon from '~/assets/rounded-success.svg?react';
+import ErrorIcon from '~/assets/rounded-warning.svg?react';
+import {
+    BackendError,
+    GuardianStatus,
+    GuardiansState,
+    ValidatedProposalType,
+    generateSecretShare,
+    registerGuardianKey,
+    registerGuardianShares,
+} from '~/shared/ffi';
+import { CCD_SYMBOL, sleep, useCountdown } from 'shared/util';
+import { ElectionPhase, SetupStep, electionConfigAtom, electionStepAtom, guardiansStateAtom } from '~/shared/store';
+
+const enum ActionStep {
+    Generate,
+    ApproveTransaction,
+    UpdateConctract,
+    Done,
+}
+
+type StepProps = PropsWithChildren<{
+    activeStep: ActionStep;
+    step: ActionStep;
+    error?: string;
+    note?: string;
+    warn?: boolean;
+}>;
+
+const enum StepStatus {
+    Inactive,
+    Active,
+    Success,
+    Error,
+}
```

Comment:
Could you please document/explain the meaning of these statuses (and actionstep above as well).